### PR TITLE
Non-existing url bug

### PR DESCRIPTION
**Pull request by sumitra19jha**
**What kind of change does this PR introduce?**
Bug and enhancement. The PR make sure that non-existing Url when entered will not proceed to login/signup page as per issue #104 

**Did you add tests for your changes?**
Yes. Below is the attached video

https://user-images.githubusercontent.com/35868598/111226541-49717f00-8607-11eb-944c-4950c8b64a34.mp4

**If relevant, did you update the documentation?**
No no need to change the documentation.

**Summary**

**Does this PR introduce a breaking change?**
After this change non-responsive url will not be called.

### CODE REVIEW

**Comment by sumitra19jha** [comment]
There is a minor error in this PR which I am fixing,

**Review by cyber-venom003 — COMMENTED** [review]
LGTM 👍

File: lib/views/pages/login_signup/login_page.dart [hunk 1]
```diff
@@ -367,15 +371,48 @@ class _LoginScreenState extends State<LoginPage> with TickerProviderStateMixin {
...
                                         _formKey.currentState.save();
-                                        setAPIURL();
-                                        _setURL();
+
+                                        setState(() {
+                                          isUrlCalled = true;
+                                        });
+
+                                        try {
+                                          final response = await http.get(
```
Comment (arteevraina) [inline lib/views/pages/login_signup/login_page.dart:395]: I would suggest handling this `async `task in a separate function for making the code testable. But, wait for the mentor's call also. They would surely have a better approach to this.

**Review by arteevraina — CHANGES_REQUESTED** [review]
Just a minor refactor required.

**Comment by sumitra19jha** [comment]
@arteevraina Sure will look into this. The main issue is that the keyword `talawa` and `graphql` is already written in the code itself. It's not the user who decides the entire url and also backend doesn't know how to handle simple static get request since the api call is a graph query. So I am facing a bit trouble, but i am looking into coming with a good solution.

**Comment by sumitra19jha** [comment]
Mentors the PR is now-ready to be merged.
Following test case is satisfied

https://user-images.githubusercontent.com/35868598/111296677-bff79680-8672-11eb-85fc-1ff6ab3e3b38.mp4

**Comment by sumitra19jha** [comment]
@Sagar2366 Conflict has been resolved here

File: lib/views/pages/login_signup/login_page.dart [hunk 2]
```diff
@@ -553,4 +560,27 @@ class _LoginScreenState extends State<LoginPage> with TickerProviderStateMixin {
...
+    try {
+      await http.get('${dropdownValue.toLowerCase()}://${urlController.text}/');
+
+      setAPIURL();
+      _setURL();
+    } catch (e) {
+      _scaffoldkey.currentState.showSnackBar(
+        SnackBar(
+            behavior: SnackBarBehavior.floating,
+            content: Text('Url does not exist')),
```
Comment (Sagar2366) [inline lib/views/pages/login_signup/login_page.dart:578]: Update message "Incorrect organization entered" or something more meaningful.
Make snackbar look good to go with current UI/UX. Currently snackbar looking very simple.

Comment (sumitra19jha) [reply lib/views/pages/login_signup/login_page.dart:578]: Sure working on it

File: lib/views/pages/login_signup/login_page.dart [hunk 3]
```diff
@@ -566,4 +590,51 @@ class _LoginScreenState extends State<LoginPage> with TickerProviderStateMixin {
...
+
+  Future<void> checkAndSetUrl() async {
+    setState(() {
+      isUrlCalled = true;
+    });
+
+    try {
+      await http.get('${dropdownValue.toLowerCase()}://${urlController.text}/');
+
+      setAPIURL();
```
Comment (Sagar2366) [inline lib/views/pages/login_signup/login_page.dart:630]: setAPIURL() method not exists, throwing error.

Screenshot 1942-12-27 at 7 53 55 PM

**Review by Sagar2366 — APPROVED** [review]

**Pull request merged**